packages/jest-environment/src/index.ts

```diff
@@ -172,6 +172,12 @@ export interface Jest {
    * local module state doesn't conflict between tests.
    */
   isolateModules(fn: () => void): Jest;
+  /**
+   * `jest.isolateModulesAsync()` is the equivalent of `jest.isolateModules()`, but for
+   * async functions to be wrapped. The caller is expected to `await` the completion of
+   * `isolateModulesAsync`.
+   */
+  isolateModulesAsync(fn: () => void): Promise<Jest>;
```

Comment:
Just thinking out loud, perhaps typings should require the callback to be async?

```suggestion
  isolateModulesAsync(fn: () => Promise<void>): Promise<Jest>;
```

Reply:
Done in 5810116f02db4f5b0ca78a0f49be40de0f925a1d